backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java

```diff
@@ -96,114 +97,127 @@ public void submitWaitlistRequest(@Valid @RequestBody WaitlistRequest body) thro
-    List<String> testingDeviceIds =
-        testingDevicesSubmitted.stream()
-            .map(
-                d -> {
-                  String deviceId =
-                      Optional.ofNullable(deviceNamesToIds.get(d)).orElse(deviceModelsToIds.get(d));
-                  if (deviceId == null) {
-                    throw new RuntimeException(
-                        String.format("Submitted device=%s not registered in DB.", d));
-                  }
-                  return deviceId;
-                })
-            .collect(Collectors.toList());
-    String defaultTestingDeviceId =
-        Optional.ofNullable(deviceNamesToIds.get(reqVars.get("defaultTestingDevice")))
-            .orElse(deviceModelsToIds.get(reqVars.get("defaultTestingDevice")));
-    if (defaultTestingDeviceId == null) {
-      throw new RuntimeException(
+    try {
+      String subject = "New account request";
+      if (LOG.isInfoEnabled()) {
+        LOG.info("Account request submitted: {}", objectMapper.writeValueAsString(body));
+      }
+
+      Map<String, String> reqVars =
+          body.toTemplateVariables().entrySet().stream()
+              .collect(
+                  HashMap::new,
+                  (m, e) ->
+                      m.put(e.getKey(), e.getValue() == null ? null : e.getValue().toString()),
+                  HashMap::putAll);
+
+      List<DeviceType> devices = _dts.fetchDeviceTypes();
+      Map<String, String> deviceNamesToIds =
+          devices.stream()
+              .collect(Collectors.toMap(d -> d.getName(), d -> d.getInternalId().toString()));
+      Map<String, String> deviceModelsToIds =
+          devices.stream()
+              .collect(Collectors.toMap(d -> d.getModel(), d -> d.getInternalId().toString()));
+
+      List<String> testingDevicesSubmitted =
+          new ArrayList<>(Arrays.asList(reqVars.get("testingDevices").split(", ")));
+      testingDevicesSubmitted.removeIf(d -> d.toLowerCase().startsWith("other"));
+      List<String> unregisteredTestingDevices = new ArrayList<String>();
+      List<String> testingDeviceIds =
+          testingDevicesSubmitted.stream()
+              .map(
+                  d -> {
+                    String deviceId =
+                        Optional.ofNullable(deviceNamesToIds.get(d))
+                            .orElse(deviceModelsToIds.get(d));
+                    if (deviceId == null) {
+                      unregisteredTestingDevices.add(d);
+                    }
+                    return deviceId;
+                  })
+              .collect(Collectors.toList());
+      // Can't easily catch a thrown exception in a stream, so move the throw outside of the stream:
+      if (!unregisteredTestingDevices.isEmpty()) {
+        throw new IOException(
+            String.format(
+                "Submitted device=%s not registered in DB.",
+                unregisteredTestingDevices.iterator().next()));
+      }
```

Comment:
Other than wrapping this whole method in a try/catch, this section is the only change (moving the null check outside of the stream).